platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/CryptographyHolder.java

```diff
@@ -48,30 +54,44 @@ private CryptographyHolder() {
     }
 
     /**
-     * By calling the method the interal {@link Cryptography} instance will be removed and a new instance will be
-     * created when {@link #get()} is called the next time.
+     * Setup the config holder.
+     *
+     * @param cryptography the {@link CryptoEngine} to use
      */
-    public static void reset() {
+    public static void set(@NonNull final Cryptography cryptography) {
         try (final Locked ignored = lock.lock()) {
-            cryptography = null;
+            CryptographyHolder.cryptography = cryptography;
         }
     }
 
+    /**
+     * Setup cryptography. Intended only to be invoked by unit tests.
+     */
+    private static void defaultTestSetup() {
```

Comment:
The method is called by the get() method. Therefore it is not called by unit test but by logic that is called everywhere. Let's name it init/reset and add info to the javadoc that it is only needed to make it useable in tests. 
Question: How long will the holder stay? Maybe we introduce a Unit extension (see https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/docs/base/test-support/test-support.md) that resets Config Holder & Crypto Holder.

Reply:
>  Let's name it init/reset and add info to the javadoc that it is only needed to make it useable in tests.

done

> Question: How long will the holder stay? 

Hopefully not long. I don't like it. Static access to anything in the context == 🤮 

> Maybe we introduce a Unit extension (see https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/docs/base/test-support/test-support.md) that resets Config Holder & Crypto Holder.

Let's discuss this the next time we are both online. Sounds like a good idea, but I'm not sure exactly what needs to go into making this happen.